src/bentoml/_internal/io_descriptors/json.py

```diff
@@ -192,6 +242,10 @@ def __init__(
                 pydantic_model, pydantic.BaseModel
             ), "'pydantic_model' must be a subclass of 'pydantic.BaseModel'."
 
+        if typeddict and is_typeddict(typeddict) is False:
```

Comment:
nit: prefer explicit check

```suggestion
        if typeddict is not None and not is_typeddict(typeddict):
```